src/skin/skinloader.cpp

```diff
@@ -213,5 +233,117 @@ SkinPointer SkinLoader::skinFromDirectory(const QDir& dir) const {
+    m_pShowSpinnyAndOrCover->setReadOnly();
+    // This is 1 if [Skin],show_spinny_cover == 1 AND [Skin],select_big_spinny_coverart == 0
+    m_pShowSmallSpinnyCover = new ControlPushButton(
+            ConfigKey("[Skin]", "show_small_spinny_or_cover"));
+    m_pShowSmallSpinnyCover->setButtonMode(ControlPushButton::TOGGLE);
+    m_pShowSmallSpinnyCover->setReadOnly();
+    // This is 1 if [Skin],show_spinny_cover == 1 AND [Skin],select_big_spinny_coverart == 1
+    m_pShowBigSpinnyCover = new ControlPushButton(ConfigKey("[Skin]", "show_big_spinny_or_cover"));
+    m_pShowBigSpinnyCover->setButtonMode(ControlPushButton::TOGGLE);
+    m_pShowBigSpinnyCover->setReadOnly();
+
+    connect(m_pShowSpinny,
+            &ControlObject::valueChanged,
+            this,
+            &SkinLoader::updateSpinnyCoverControls);
+    connect(m_pShowCover,
+            &ControlObject::valueChanged,
+            this,
+            &SkinLoader::updateSpinnyCoverControls);
+    connect(m_pSelectBigSpinnyCover,
+            &ControlObject::valueChanged,
+            this,
+            &SkinLoader::updateSpinnyCoverControls);
+
+    m_spinnyCoverControlsCreated = true;
+    updateSpinnyCoverControls();
+}
+
+void SkinLoader::updateSpinnyCoverControls() {
+    if (!m_spinnyCoverControlsCreated) {
+        return;
+    }
+    m_pShowSpinnyAndOrCover->setAndConfirm(
+            (m_pShowSpinny->toBool() || m_pShowCover->toBool())
+                    ? 1.0
+                    : 0.0);
+    m_pShowSmallSpinnyCover->setAndConfirm(
+            (m_pShowSpinnyAndOrCover->toBool() && !m_pSelectBigSpinnyCover->toBool())
+                    ? 1.0
+                    : 0.0);
+    m_pShowBigSpinnyCover->setAndConfirm(
+            (m_pShowSpinnyAndOrCover->toBool() &&
+                    m_pSelectBigSpinnyCover->toBool())
+                    ? 1.0
+                    : 0.0);
+}
+
+void SkinLoader::setupMicDuckingControls() {
+    if (m_micDuckingControlsCreated) {
+        return;
+    }
+    // This is 1 if at least one microphone device is configured
+    m_pShowDuckingControls = new ControlPushButton(ConfigKey("[Skin]", "show_ducking_controls"));
+    m_pShowDuckingControls->setButtonMode(ControlPushButton::TOGGLE);
+    m_pShowDuckingControls->setReadOnly();
+
+    m_pNumMics = new ControlProxy("[Master]", "num_microphones", this);
+    m_pNumMics->connectValueChanged(this, &SkinLoader::slotNumMicsChanged);
+
+    m_micDuckingControlsCreated = true;
+    slotNumMicsChanged(m_pNumMics->get());
+}
+
+void SkinLoader::slotNumMicsChanged(double dNumMics) {
+    int numMics = static_cast<int>(dNumMics);
+
+    if (numMics <= m_numMicsEnabled) {
+        return;
+    }
+
+    for (int micNum = m_numMicsEnabled; micNum <= numMics; ++micNum) {
+        QString micGroup = PlayerManager::groupForMicrophone(micNum - 1);
+        ControlProxy* pMicEnabled = new ControlProxy(micGroup, "input_configured", this);
+        m_pMicConfiguredControls.push_back(pMicEnabled);
```

Comment:
This may become a `std::vector<PollingControlProxy>` Than this boils down to: 
```suggestion
        m_pMicConfiguredControls.emplace_back(micGroup, "input_configured");
```
This way the PollingControlProxy is crated directly in the vector's memory without any copy or move. 
PollingControlProxy works here in contrast to ControlProxy, because you do not need signals.

Reply:
Thanks for the vector idea.
Though, we need ControlProxy here because I want to catch any mic configuration change and adjust the duckingGUI control